src/App.tsx

```diff
@@ -124,12 +129,26 @@ function App() {
     }
   };
 
+  const setExtensionIdToLocalStorage = useCallback(
+    (_extensionId: string): void => {
+      const walletList = Object.values(walletMapEntries);
+      const detectedExtensionIds = walletList.map(
+        (entry) => entry.params.extensionId,
+      );
+      if (detectedExtensionIds.includes(_extensionId)) {
+        return;
+      }
+      localStorage.setItem('extensionId', _extensionId);
+    },
+    [walletMapEntries],
+  );
+
   const handleConnect = () => {
-    if (extensionId && provider) {
+    if (provider && extensionId) {
       try {
         const connected = provider.connect(extensionId);
-        setisExternallyConnectableConnected(connected);
-        localStorage.setItem('extensionId', extensionId);
+        setIsExternallyConnectableConnected(connected);
+        setExtensionIdToLocalStorage(extensionId);
```

Comment:
The reasoning for this more complex logic for setting an item in localStorage is that there is a useEffect that gets this item _FROM_ localStorage and automatically sets the app state to connected, so I can’t interact with the button or input box for `extensionId`.
Rather than put new state for tracking connection state, thought this was a simpler approach (less code overall at the end of it)